src/converters/fromZigbee.ts

```diff
@@ -1921,35 +1921,105 @@ const converters1 = {
     elko_thermostat: {
         cluster: 'hvacThermostat',
         type: ['attributeReport', 'readResponse'],
+        options: [exposes.options.local_temperature_based_on_sensor()],
         convert: (model, msg, publish, options, meta) => {
             const result: KeyValueAny = {};
             const data = msg.data;
-
+            if (data.localTemp !== undefined) {
+                const value = precisionRound(msg.data['localTemp'], 2) / 100;
+                const valuesFloorSensor = ['floor', 'supervisor_floor'];
+                const sensorType = globalStore.getValue(msg.endpoint, 'sensor');
```

Comment:
It would be better to get the value from the `meta.state`, the `globalStore` is always reset after every z2m restart.

Reply:
Thank you for the valuable input. I agree (now that I understand the meta.state capability) that this is much better and I've made the change.